mithril-signer/src/services/upkeep_service.rs

```diff
@@ -42,16 +56,40 @@ impl SignerUpkeepService {
         main_db_connection: Arc<SqliteConnection>,
         cardano_tx_connection_pool: Arc<SqliteConnectionPool>,
         signed_entity_type_lock: Arc<SignedEntityTypeLock>,
+        pruning_tasks: Vec<Arc<dyn EpochPruningTask>>,
+        store_retention_limit: Option<u64>,
         logger: Logger,
     ) -> Self {
         Self {
             main_db_connection,
             cardano_tx_connection_pool,
             signed_entity_type_lock,
+            pruning_tasks,
+            store_retention_limit,
             logger,
         }
     }
 
+    async fn execute_pruning_tasks(&self, current_epoch: Epoch) -> StdResult<()> {
```

Comment:
I'm not sure that we need to support multiple pruning tasks here. 
Also this design does not seem very adapted if the epoch thresholds are different from one task to another.

Reply:
Good point for the threshold, I will move the computation to the tasks themself and only pass the current epoch.

For the multiple tasks I was thinking we could move pruning on our current tasks to that kind of design, this would avoid doing it every-time we save data.